### PR TITLE
feat: add unit test for app store

### PR DESCRIPTION
**Pull request by nikaaru**
# Summary

**vitest** is configured to work with **zustand** And a series of unit tests for the App Store have been added to the widget

# How did you test this change?

You can run the `yarn test` command and all the tests should pass correctly.

To see more details, you can enter the embedded project with the `cd widget/embedded/` command and run the `yarn test ` command again.

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Implemented a user interface (UI) change, referencing our Figma design to ensure pixel-perfect precision.

### CODE REVIEW

**Review by yeager-eren — CHANGES_REQUESTED** [review]

**Comment by yeager-eren** [comment]
I've changed your original code to keep only those who should be kept and added another tool to generate fake data. I didn't know what exactly should be result of some of the tests, i made them `skip`, please make sure you will check them. I also put some `TODO:` and `.todo` please make sure you won't miss them.

Besides of this PR, I suggest to take a look at (your previous PRs)[https://github.com/rango-exchange/rango-client/pull/634] and make sure you've covered all the cases mentions in that PR.

File: package.json [hunk 1]
```diff
@@ -47,8 +47,8 @@
     "i18n:extract": "lingui extract --clean",
     "i18n:compile": "lingui compile --typescript",
     "i18n": "yarn i18n:extract && yarn i18n:compile",
-    "test": "vitest",
-    "test:coverage": "vitest run --coverage",
+    "test": "nx run-many --target=test",
+    "test:coverage": "nx run-many --target=test:coverage",
```
[the remark this answers is not in the document]
Comment (nikaaru) [reply package.json]: I used the **vitest** command and it worked.

**Comment by nikaaru** [comment]
> I've changed your original code to keep only those who should be kept and added another tool to generate fake data. I didn't know what exactly should be result of some of the tests, i made them `skip`, please make sure you will check them. I also put some `TODO:` and `.todo` please make sure you won't miss them.
> 
> Besides of this PR, I suggest to take a look at (your previous PRs)[https://github.com/[/pull/634](https://github.com/rango-exchange/rango-client/pull/634)] and make sure you've covered all the cases mentions in that PR.

I fixed all the todos.
But I didn't understand what is your point about this line:

`    // TODO: write a test util to update config.
`
I think it will be cleaner to update the config in the test file.

File: widget/embedded/src/test-utils/constants.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
+export const TOKENS_COUNT = 20;
```
Comment (yeager-eren) [inline widget/embedded/src/test-utils/constants.ts:1]: Let's keep related pieces together. Move back this to `fixtures`. Constants doesn't make any sense in context of `test utils` and it's part of `fixture` actually.

**Comment by yeager-eren** [comment]
> > I've changed your original code to keep only those who should be kept and added another tool to generate fake data. I didn't know what exactly should be result of some of the tests, i made them `skip`, please make sure you will check them. I also put some `TODO:` and `.todo` please make sure you won't miss them.
> > Besides of this PR, I suggest to take a look at (your previous PRs)[https://github.com/[/pull/634](https://github.com/rango-exchange/rango-client/pull/634)] and make sure you've covered all the cases mentions in that PR.
> 
> I fixed all the todos. But I didn't understand what is your point about this line:
> 
> ` // TODO: write a test util to update config.`
> 
> I think it will be cleaner to update the config in the test file.

I meant creating a `test-utils/store.ts` and the wrtie a util called `updateConfig`  to be used whenever we need to update config.
Instead of this:
```ts
   appStoreState.config.from = {
      ...appStoreState.config.from,
      pinnedTokens: [
        {
          symbol: rangoToken.symbol,
          address: rangoToken.address,
          blockchain: rangoToken.blockchain,
        },
      ],
    };
```
It will be like:
```ts
updateConfig(appStoreState, {
 from: {
    pinnedTokens: [ ....]
 }
})
```

File: widget/embedded/src/store/slices/data.test.ts [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+  initData._blockchainsMapByName.set(rangoBlockchain.name, rangoBlockchain);
+  customTokens.forEach((token) => initData._tokens.push(token));
+
+  const appStore = createAppStore();
+  appStore.setState(initData);
+  appStoreState = appStore.getState();
+});
+
+describe('check sorting tokens is working correctly in app store', () => {
+  test('put pinned tokens first', async () => {
```
Comment (yeager-eren) [inline widget/embedded/src/store/slices/data.test.ts:58]: Please remove `async` if they aren't needed.

File: widget/embedded/src/test-utils/constants.ts (hunk 2 above)
Comment (nikaaru) [reply widget/embedded/src/test-utils/constants.ts:1]: fixed

**Review by yeager-eren — APPROVED** [review]

File: widget/embedded/src/setup-vitest.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import { vi } from 'vitest';
+
+// to make it works like Jest (auto-mocking)
+vi.mock('zustand');
```
[the remark this answers is not in the document]
Comment (nikaaru) [reply widget/embedded/src/setup-vitest.ts]: Done

**Pull request merged**